opencensus/proto/trace/v1/trace.proto

```diff
@@ -47,22 +47,31 @@ message Span {
   // This field is required.
   bytes span_id = 2;
 
-  // The `tracestate` field conveys information about request position in multiple distributed
-  // tracing graphs.
-  //
-  // There can be a maximum of 32 members in the map.
-  //
-  // The key must begin with a lowercase letter, and can only contain lowercase letters 'a'-'z',
-  // digits '0'-'9', underscores '_', dashes '-', asterisks '*', and forward slashes '/'. For
-  // multi-tenant vendors scenarios '@' sign can be used to prefix vendor name. The maximum length
-  // for the key is 256 characters.
-  //
-  // The value is opaque string up to 256 characters printable ASCII RFC0020 characters (i.e., the
-  // range 0x20 to 0x7E) except ',' and '='. Note that this also excludes tabs, newlines, carriage
-  // returns, etc.
+  // This field conveys information about request position in multiple distributed tracing graphs.
+  // It is a list of Tracestate.Entry with a maximum of 32 members in the list.
   //
   // See the https://github.com/w3c/distributed-tracing for more details about this field.
-  map<string, string> tracestate = 15;
+  message Tracestate {
+    message Entry {
+      // The key must begin with a lowercase letter, and can only contain
+      // lowercase letters 'a'-'z', digits '0'-'9', underscores '_', dashes
+      // '-', asterisks '*', and forward slashes '/'. For multi-tenant vendors
+      // scenarios '@' sign can be used to prefix vendor name. The maximum
+      // length for the key is 256 characters.
+      string key = 1;
+
+      // The value is opaque string up to 256 characters printable ASCII
+      // RFC0020 characters (i.e., the range 0x20 to 0x7E) except ',' and '='.
+      // Note that this also excludes tabs, newlines, carriage returns, etc.
+      string value = 2;
+    }
+
+    // A list of entries that represent the Tracestate.
+    repeated Entry entries = 1;
+  }
+
+  // The Tracestate on the span.
+  Tracestate tracestate = 15;
```

Comment:
another option is to have a string and not parse it except to remove your incoming tracestate. OTOH what you have is more flexible for future changes. As mentioned in the spec review "For multi-tenant vendors scenarios '@' sign can be used to prefix vendor name." is highly speculative, and the format isn't well thought through. Usually '@' is not a character promoted for use for various reasons I can think of a few. I wouldn't advertise this in good faith.

Reply:
ps option for ridding `@` is just deleting the advice (ideally also from where it was copy/pasted from). We should not encourage characters that routinely require escaping.